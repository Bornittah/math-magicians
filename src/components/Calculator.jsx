import React from 'react';
import Button from './shared/Button/Button';
import calculate from './logic/calculate';
import './style.css';

class Calculator extends React.Component {
  constructor(props) {
    super(props);
    this.state = {};
    this.onClick = this.onClick.bind(this);
  }

  onClick = (e) => {
    this.setState((data) => calculate(data, e.target.value));
  };

  render() {
    const { next, total } = this.state;
    return (
      <div className="wrapper flex align-center justify-center">
        <div className="calculator">
          <div className="answer-display">
            <p>{next || total || 0}</p>
          </div>
          <div className="row flex space-btn">
            <Button
              className="button gray-bg flex-1"
              value="AC"
              onClick={this.onClick}
            />
            <Button
              className="button gray-bg flex-1"
              value="+/-"
              onClick={this.onClick}
            />
            <Button
              className="button gray-bg flex-1"
              value="%"
              onClick={this.onClick}
            />
            <Button
              className="button orange-bg flex-1"
              value="&#247;"
              onClick={this.onClick}
            />
          </div>
          <div className="row flex space-btn">
            <Button
              className="button gray-bg flex-1"
              value="7"
              onClick={this.onClick}
            />
            <Button
              className="button gray-bg flex-1"
              value="8"
              onClick={this.onClick}
            />
            <Button
              className="button gray-bg flex-1"
              value="9"
              onClick={this.onClick}
            />
            <Button
              className="button orange-bg flex-1"
              value="x"
              onClick={this.onClick}
            />
          </div>
          <div className="row flex space-btn">
            <Button
              className="button gray-bg flex-1"
              value="4"
              onClick={this.onClick}
            />
            <Button
              className="button gray-bg flex-1"
              value="5"
              onClick={this.onClick}
            />
            <Button
              className="button gray-bg flex-1"
              value="6"
              onClick={this.onClick}
            />
            <Button
              className="button orange-bg flex-1"
              value="-"
              onClick={this.onClick}
            />
          </div>
          <div className="row flex space-btn">
            <Button
              className="button gray-bg flex-1"
              value="1"
              onClick={this.onClick}
            />
            <Button
              className="button gray-bg flex-1"
              value="2"
              onClick={this.onClick}
            />
            <Button
              className="button gray-bg flex-1"
              value="3"
              onClick={this.onClick}
            />
            <Button
              className="button orange-bg flex-1"
              value="+"
              onClick={this.onClick}
            />
          </div>
          <div className="row flex space-btn">
            <Button
              className="button gray-bg flex-2"
              value="0"
              onClick={this.onClick}
            />
            <Button
              className="button gray-bg flex-1 "
              value="."
              onClick={this.onClick}
            />
            <Button
              className="button orange-bg flex-1 "
              value="="
              onClick={this.onClick}
            />
          </div>
        </div>
      </div>
    );
  }
}

export default Calculator;
