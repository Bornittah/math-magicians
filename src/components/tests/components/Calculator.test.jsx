import renderer from 'react-test-renderer';
import Calculator from '../../Calculator';

describe('Testing Calculator component', () => {
  it('Testing with snapshot', () => {
    const tree = renderer.create(<Calculator />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
