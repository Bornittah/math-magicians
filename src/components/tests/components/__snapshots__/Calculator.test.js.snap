// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Calculator component Testing with snapshot 1`] = `
<div
  className="wrapper flex align-center justify-center"
>
  <div
    className="calculator"
  >
    <div
      className="answer-display"
    >
      <p>
        0
      </p>
    </div>
    <div
      className="row flex space-btn"
    >
      <button
        className="button gray-bg flex-1"
        onClick={[Function]}
        type="button"
        value="AC"
      >
        AC
      </button>
      <button
        className="button gray-bg flex-1"
        onClick={[Function]}
        type="button"
        value="+/-"
      >
        +/-
      </button>
      <button
        className="button gray-bg flex-1"
        onClick={[Function]}
        type="button"
        value="%"
      >
        %
      </button>
      <button
        className="button orange-bg flex-1"
        onClick={[Function]}
        type="button"
        value="รท"
      >
        รท
      </button>
    </div>
    <div
      className="row flex space-btn"
    >
      <button
        className="button gray-bg flex-1"
        onClick={[Function]}
        type="button"
        value="7"
      >
        7
      </button>
      <button
        className="button gray-bg flex-1"
        onClick={[Function]}
        type="button"
        value="8"
      >
        8
      </button>
      <button
        className="button gray-bg flex-1"
        onClick={[Function]}
        type="button"
        value="9"
      >
        9
      </button>
      <button
        className="button orange-bg flex-1"
        onClick={[Function]}
        type="button"
        value="x"
      >
        x
      </button>
    </div>
    <div
      className="row flex space-btn"
    >
      <button
        className="button gray-bg flex-1"
        onClick={[Function]}
        type="button"
        value="4"
      >
        4
      </button>
      <button
        className="button gray-bg flex-1"
        onClick={[Function]}
        type="button"
        value="5"
      >
        5
      </button>
      <button
        className="button gray-bg flex-1"
        onClick={[Function]}
        type="button"
        value="6"
      >
        6
      </button>
      <button
        className="button orange-bg flex-1"
        onClick={[Function]}
        type="button"
        value="-"
      >
        -
      </button>
    </div>
    <div
      className="row flex space-btn"
    >
      <button
        className="button gray-bg flex-1"
        onClick={[Function]}
        type="button"
        value="1"
      >
        1
      </button>
      <button
        className="button gray-bg flex-1"
        onClick={[Function]}
        type="button"
        value="2"
      >
        2
      </button>
      <button
        className="button gray-bg flex-1"
        onClick={[Function]}
        type="button"
        value="3"
      >
        3
      </button>
      <button
        className="button orange-bg flex-1"
        onClick={[Function]}
        type="button"
        value="+"
      >
        +
      </button>
    </div>
    <div
      className="row flex space-btn"
    >
      <button
        className="button gray-bg flex-2"
        onClick={[Function]}
        type="button"
        value="0"
      >
        0
      </button>
      <button
        className="button gray-bg flex-1 "
        onClick={[Function]}
        type="button"
        value="."
      >
        .
      </button>
      <button
        className="button orange-bg flex-1 "
        onClick={[Function]}
        type="button"
        value="="
      >
        =
      </button>
    </div>
  </div>
</div>
`;
